plugins/gossip/autopeering.go

```diff
@@ -46,27 +48,31 @@ func configureAutopeering() {
 		if !ev.Status {
 			return // ignore rejected peering
 		}
-		gossipAddr := ev.Peer.Services().Get(services.GossipServiceKey()).String()
-		apLog.Infof("[incoming peering] whitelisting %s / %s / %s", ev.Peer.Address(), gossipAddr, ev.Peer.ID())
+		gossipAddr := ev.Peer.Services().Get(services.GossipServiceKey())
+		gossipAddrStr := net.JoinHostPort(ev.Peer.IP().String(), strconv.Itoa(gossipAddr.Port()))
+		apLog.Infof("[incoming peering] whitelisting %s / %s / %s", ev.Peer.Address(), gossipAddrStr, ev.Peer.ID())
+
 		// whitelist the given peer
 		neighborsLock.Lock()
 		defer neighborsLock.Unlock()
+
 		// will be grabbed later by the incoming connection
-		allowedIdentities[gossipAddr] = ev.Peer
+		allowedIdentities[gossipAddrStr] = ev.Peer
+
 		// remove from host blacklist
 		hostsBlacklistLock.Lock()
-		originAddr, _ := iputils.ParseOriginAddress(gossipAddr)
-		delete(hostsBlacklist, originAddr.Addr)
+		delete(hostsBlacklist, ev.Peer.Address().IP.String())
```

Comment:
Why `ev.Peer.Address().IP` instead of `ev.Peer.IP().String()`?
Before exactly the IP from the gossip service was used, this would correspond to `ev.Peer.IP()`